### DISCUSSION TITLE
Issue in running mhm model

### DISCUSSION THREAD

**Discussion by wk1984** [mhm-ufz/mHM#86] [Q&A] [answered]
I prepared input files and *.nml file for running my test case. When I run the model, it seems to stop at a step (see attached figure).
![WX20240123-141727@2x](https://github.com/mhm-ufz/mHM/assets/7428153/5e879f2a-df33-44cc-830c-c33a67828aa9)

I also attached all files here for further discussion and checking.

[moselle_domain.zip](https://github.com/mhm-ufz/mHM/files/14019138/moselle_domain.zip)

**Comment by ViniciusBMoreira** [comment]
Hey! I think that you just don't have these geological units in the corresponding .asc file of your geological .txt file.

**Reply by wk1984** [reply]
Hey @BryanSouzaM, thank you for replying. But I think it may not solve the issue because I already set all pixels to geological unit = 1. For double checking, I now remove the units (2-10) in geological.txt and mhm_parameters.nml. The issue is still the same. Confusing for me.

**Reply by ViniciusBMoreira** [reply]
I think that the Geological message is just an Warning.

Maybe you need to check the 169 outlets of the model?

Also, the gauge is not coincident with the facc.

![image](https://github.com/mhm-ufz/mHM/assets/61054098/0ef8c33f-9f38-488f-b22e-22c3ac5eb24e)

Additionally, your gauges folder is empty.

But anyway, the error may not stem from any of my comments.

But at the moment when your model froze, the mHM is operating the mRM (opening dem, fdir and facc, maybe?):

My model on the left for comparison:

![image](https://github.com/mhm-ufz/mHM/assets/61054098/da3e4e51-67eb-4cc1-bcfc-ebba3688caed)

**Reply by ViniciusBMoreira** [reply, marked as answer]
@wk1984 

Checking your data, I noticed that the flow direction (fdir) is incorrectly oriented. "1" should be towards the right, and yours is towards the southeast.

![image](https://github.com/mhm-ufz/mHM/assets/61054098/840d767e-d4b4-4412-a41f-58695a7bff6e)

![image](https://github.com/mhm-ufz/mHM/assets/61054098/c4d7adc8-8b07-4a3f-8f91-355ff35f92b8)

**Reply by wk1984** [reply]
Yeah, it works when I fix the fdir issue. Many thanks for your help.

**Comment by cuneyd** [comment]
Hello,

I have a similar issue in my setup for Cyprus island. Your comments would be very much appreciated.

Error message in mhm:
Something went wrong during L11_link_location, movedownonecell got stuck in infinite loop at cell (      1328        155


![image_2024-03-28_130744000](https://github.com/mhm-ufz/mHM/assets/7171087/88ea6494-c42f-4860-accf-f05aa6e59446)

facc fdir and idgauges files in a zip folder:
[fdir_facc_cyprus.zip](https://github.com/mhm-ufz/mHM/files/14787606/fdir_facc_cyprus.zip)

Thanks in advance,
Cüneyd

**Reply by ViniciusBMoreira** [reply]
Hey Cüneyd!

I think you should open a new discussion for posterior consulting.

Try this fdir.asc (It's just a test, it might not solve the problem. But it replaces the zero values with -9999.)

[fdir.zip](https://github.com/mhm-ufz/mHM/files/14791307/fdir.zip)

But looking at your file, it seems that this problematic area is not important for the simulation, am I wrong? Maybe using a mask to remove it would work...

And again, maybe you should start a new discussion since this one has been closed.

Vinicius Bryan
